src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,99 @@
+#include <hyro/SignalGeneratorComponent.h>
+
+using namespace hyro;
+
+namespace signal_generator
+{
+
+std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+
+hyro::Result
+SignalGeneratorComponent::init(const hyro::ComponentConfiguration &config)
+{
+  m_signal.setStep(5);
+
+  m_signal_output = this->registerOutput<hyro::SignalMsgs>("analog_output"_uri, config);
+
+  std::shared_ptr<ChannelOutput<std::vector<int>>>
+  m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, config);
+
```

Comment:
As explained in the `DigitalGeneratorComponent::init`, you need to initialize the parameters using the function ` config.parameters.getParameter`.